src/Maestro/maestro-angular/src/app/page/build/build.component.ts

```diff
@@ -311,6 +334,30 @@ export class BuildComponent implements OnInit, OnChanges {
       );
   }
 
+  public getOngoingBuildInfo(build: Build): Observable<string | null> {
+    if (!build.azureDevOpsAccount) {
+      throw new Error("azureDevOpsAccount undefined");
+    }
+    if (!build.azureDevOpsProject) {
+      throw new Error("azureDevOpsProject undefined");
+    }
+    if (!build.azureDevOpsBuildDefinitionId) {
+      throw new Error("azureDevOpsBuildDefinitionId undefined");
+    }
+    if (!build.azureDevOpsBranch) {
+      throw new Error("azureDevOpsBranch undefined");
+    }
+    return this.buildStatusService.getBranchStatus(build.azureDevOpsAccount, build.azureDevOpsProject, build.azureDevOpsBuildDefinitionId, build.azureDevOpsBranch, 1, "inProgress")
+    .pipe(
+      map(builds => {
+        if (builds.count > 0) {
+          return builds.value[0]["_links"]["web"]["href"];
```

Comment:
I am not ts expert but would builds.value[0]._links.web.href; works better? Why don't we use BuildStatusInProgress  interface here (retyping).